### ISSUE TITLE
GraphFactory could not instantiate this Graph implementation?

### ISSUE THREAD

**Issue by youzp** [rayokota/hgraphdb#6]
hi, @rayokota 
With version 0.4.1,the connection to HBase is normal when hbase.zookeeper.quorum was set one ip address, but abnormal when hbase.zookeeper.quorum was set multiple ip address.
```
HBaseGraphConfiguration cfg = new HBaseGraphConfiguration()
.setInstanceType(HBaseGraphConfiguration.InstanceType.DISTRIBUTED)
.setGraphNamespace("testgraph")
.setCreateTables(true)
.set("hbase.zookeeper.quorum", "192.168.178.91:2181,192.168.178.92:2181,192.168.178.93:2181")
.set("zookeeper.znode.parent", "/hbase_lions");

HBaseGraph graph = (HBaseGraph) GraphFactory.open(cfg);
```

errors:
```
2016-12-08 10:25:29 INFO  org.apache.zookeeper.ZooKeeper - Initiating client connection, connectString=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93:2181] sessionTimeout=180000 watcher=hconnection-0x12d3a4e90x0, quorum=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93:2181], baseZNode=/hbase_lions
Exception in thread "main" java.lang.RuntimeException: GraphFactory could not instantiate this Graph implementation [class io.hgraphdb.HBaseGraph]
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:82)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:70)
	at com.jd.risk.hbase.demo.HGraphDBDemo.main(HGraphDBDemo.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:78)
	... 7 more
Caused by: io.hgraphdb.HBaseGraphException: java.io.IOException: java.lang.reflect.InvocationTargetException
	at io.hgraphdb.HBaseGraphUtils.getConnection(HBaseGraphUtils.java:62)
	at io.hgraphdb.HBaseGraph.<init>(HBaseGraph.java:156)
	at io.hgraphdb.HBaseGraph.<init>(HBaseGraph.java:135)
	at io.hgraphdb.HBaseGraph.open(HBaseGraph.java:123)
	... 12 more
Caused by: java.io.IOException: java.lang.reflect.InvocationTargetException
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:240)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:218)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:119)
	at io.hgraphdb.HBaseGraphUtils.getConnection(HBaseGraphUtils.java:59)
	... 15 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:238)
	... 18 more
Caused by: java.lang.NumberFormatException: For input string: "2181]"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at org.apache.zookeeper.client.ConnectStringParser.<init>(ConnectStringParser.java:72)
	at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:443)
	at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:380)
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.checkZk(RecoverableZooKeeper.java:141)
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.exists(RecoverableZooKeeper.java:221)
	at org.apache.hadoop.hbase.zookeeper.ZKUtil.checkExists(ZKUtil.java:541)
	at org.apache.hadoop.hbase.zookeeper.ZKClusterId.readClusterIdZNode(ZKClusterId.java:65)
	at org.apache.hadoop.hbase.client.ZooKeeperRegistry.getClusterId(ZooKeeperRegistry.java:105)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.retrieveClusterId(ConnectionManager.java:879)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.<init>(ConnectionManager.java:635)
	... 23 more
```

**Comment by tellisnz** [comment]
I believe hbase.zookeeper.quorum should just be a comma separated list of hosts only - port is set with hbase.zookeeper.clientPort. See https://github.com/apache/hbase/blob/master/hbase-common/src/main/resources/hbase-default.xml for more info.

**Comment by rayokota** [comment]
@tellisnz is correct, thanks for answering.

**Comment by youzp** [comment]
hi,@rayokota @tellisnz 
Thanks for answering.I modified the code, but still get an error.
`set("hbase.zookeeper.quorum", "192.168.178.91,192.168.178.92,192.168.178.93") `

```
2016-12-09 09:30:47 ERROR org.apache.hadoop.hbase.zookeeper.ZooKeeperWatcher - hconnection-0x12d3a4e90x0, quorum=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93]:2181, baseZNode=/hbase_lions Received unexpected KeeperException, re-throwing exception
org.apache.zookeeper.KeeperException$OperationTimeoutException: KeeperErrorCode = OperationTimeout
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.checkZk(RecoverableZooKeeper.java:144)
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.exists(RecoverableZooKeeper.java:221)
	at org.apache.hadoop.hbase.zookeeper.ZKUtil.checkExists(ZKUtil.java:541)
	at org.apache.hadoop.hbase.zookeeper.ZKClusterId.readClusterIdZNode(ZKClusterId.java:65)
	at org.apache.hadoop.hbase.client.ZooKeeperRegistry.getClusterId(ZooKeeperRegistry.java:105)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.retrieveClusterId(ConnectionManager.java:879)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.<init>(ConnectionManager.java:635)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:238)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:218)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:119)
	at io.hgraphdb.HBaseGraphUtils.getConnection(HBaseGraphUtils.java:59)
	at io.hgraphdb.HBaseGraph.<init>(HBaseGraph.java:156)
	at io.hgraphdb.HBaseGraph.<init>(HBaseGraph.java:135)
	at io.hgraphdb.HBaseGraph.open(HBaseGraph.java:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:78)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:70)
	at com.jd.risk.hbase.demo.HGraphDBDemo.main(HGraphDBDemo.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
2016-12-09 09:30:47 WARN  org.apache.hadoop.hbase.client.ZooKeeperRegistry - Can't retrieve clusterId from Zookeeper
org.apache.zookeeper.KeeperException$OperationTimeoutException: KeeperErrorCode = OperationTimeout
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.checkZk(RecoverableZooKeeper.java:144)
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.exists(RecoverableZooKeeper.java:221)
```

**Comment by msjbear** [comment]
the same error to me, when connecting to zookeeper, the quorum passes the error format
```
the connectString=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93]:2181 
```

the detail error:
```
2016-12-09 09:45:54 WARN  org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper - Possibly transient ZooKeeper, quorum=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93]:2181, exception=org.apache.zookeeper.KeeperException$OperationTimeoutException: KeeperErrorCode = OperationTimeout
2016-12-09 09:45:55 INFO  org.apache.zookeeper.ZooKeeper - Initiating client connection, connectString=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93]:2181 sessionTimeout=180000 watcher=hconnection-0x589b36320x0, quorum=[192.168.178.91:2181, 192.168.178.92:2181, 192.168.178.93]:2181, baseZNode=/hbase_lions
2016-12-09 09:45:55 WARN  org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper - Unable to create ZooKeeper Connection
java.net.UnknownHostException: [192.168.178.91: invalid IPv6 address
	at java.net.InetAddress.getAllByName(InetAddress.java:1146)
	at java.net.InetAddress.getAllByName(InetAddress.java:1126)
	at org.apache.zookeeper.client.StaticHostProvider.<init>(StaticHostProvider.java:61)
	at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:445)
	at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:380)
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.checkZk(RecoverableZooKeeper.java:141)
	at org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper.exists(RecoverableZooKeeper.java:221)
	at org.apache.hadoop.hbase.zookeeper.ZKUtil.checkExists(ZKUtil.java:541)
	at org.apache.hadoop.hbase.zookeeper.ZKClusterId.readClusterIdZNode(ZKClusterId.java:65)
	at org.apache.hadoop.hbase.client.ZooKeeperRegistry.getClusterId(ZooKeeperRegistry.java:105)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.retrieveClusterId(ConnectionManager.java:879)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.<init>(ConnectionManager.java:635)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:238)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:218)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:119)
	at io.hgraphdb.HBaseGraphUtils.getConnection(HBaseGraphUtils.java:59)
	at io.hgraphdb.HBaseGraph.<init>(HBaseGraph.java:156)
	at io.hgraphdb.HBaseGraph.<init>(HBaseGraph.java:135)
	at io.hgraphdb.HBaseGraph.open(HBaseGraph.java:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:78)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:70)
	at com.jd.risk.hbase.demo.HGraphDBDemo.main(HGraphDBDemo.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

**Comment by rayokota** [comment]
Ok, this is fixed in 0.4.2.  Please let me know if you have any further issues.  Thanks.

**Comment by youzp** [comment]
@rayokota Thanks, now it's OK with 0.4.2.
